rustls/src/msgs/handshake.rs

```diff
@@ -1426,6 +1454,121 @@ impl CertificateEntry {
     }
 }
 
+#[derive(Debug)]
+pub struct CompressedCertificatePayload {
+    pub algorithm: CertificateCompressionAlgorithm,
+    pub uncompressed_length: codec::u24,
+    pub compressed_certificate_message: PayloadU24,
+}
+
+impl Codec for CompressedCertificatePayload {
+    fn encode(&self, bytes: &mut Vec<u8>) {
+        self.algorithm.encode(bytes);
+        self.uncompressed_length.encode(bytes);
+        self.compressed_certificate_message
+            .encode(bytes);
+    }
+
+    fn read(r: &mut Reader) -> Option<CompressedCertificatePayload> {
+        Some(CompressedCertificatePayload {
+            algorithm: CertificateCompressionAlgorithm::read(r)?,
+            uncompressed_length: codec::u24::read(r)?,
+            compressed_certificate_message: PayloadU24::read(r)?,
+        })
+    }
+}
+
+impl CompressedCertificatePayload {
+    pub fn compress_with(
```

Comment:
I'd loosely prefer to see this code live elsewhere in the crate -- the msgs module is trying to be a type-safe expression of the TLS protocol types, but without any processing/manipulation of them.

Reply:
Perhaps it should live somewhere in `crate::compression::`?